### ISSUE TITLE
fails to run on mac os x if ipv6 is configured

### ISSUE THREAD

**Issue by nkts** [sshuttle/sshuttle#74]
Error:
```
fatal: Feature ipv6 not supported with method pf.
```

**Comment by brianmay** [comment]
Sounds like IPv6 isn't getting disabled by default for pf. Try ``--disable-ipv6`` until this is fixed.

**Comment by nkts** [comment]
Dunno, I have removed one installed with brew - looks like not the latest version.
Next:
```
$ python --version
Python 3.5.0
$ pip install --user sshuttle
Collecting sshuttle
  Downloading sshuttle-0.77.2.tar.gz (63kB)
    100% |████████████████████████████████| 71kB 894kB/s
Installing collected packages: sshuttle
  Running setup.py install for sshuttle ... done
Successfully installed sshuttle-0.77.2
$ sshuttle --disable-ipv6 --dns -r pi@rpi2.lan 0/0
fatal: Feature ipv6 not supported with method pf.
```

**Comment by brianmay** [comment]
:-(

Ok, looks like you are doing the right thing. Will need to look into that.

**Comment by brianmay** [comment]
Can you please run:

```
sshuttle --disable-ipv6 --dns -vvv -r pi@rpi2.lan 0/0
```

Note the -vvv must come after the long options due to #75.

Thanks.

**Comment by nkts** [comment]
I got this:
```
$ sshuttle --disable-ipv6 --dns -vvv -r pi@rpi2.lan 0/0
Starting sshuttle proxy.
[local sudo] Password:
firewall manager: Starting firewall with Python version 3.5.0
firewall manager: ready method name pf.
fatal: Feature ipv6 not supported with method pf.
```

**Comment by brianmay** [comment]
Looks like the -vvv didn't actually do anything. Contrary to my previous advice, please try:

```
sshuttle -vvv --disable-ipv6 --dns -r pi@rpi2.lan 0/0
```

**Comment by nkts** [comment]
Above output. I could have look at code, but that won't happen till weekend.

**Comment by brianmay** [comment]
Was hoping to see what would get displayed after "IPv6 enabled:" however it looks like verbose isn't working for you. Possibly related to #75.

After reading the code again, I speculating that your /etc/resolv.conf has IPv6 entries in it; I think I see the problem now. If I am correct, dropping the --dns option should fix the immediate error.

**Comment by nkts** [comment]
Yes, without dns seems to work, tho I need dns which is on remote subnet. I have an ipv6 address in /etc/resolve.conf together with ipv4. rpi2.lan resolves to ipv6 too, but ssh is ok, to connect to it.

```
$ sshuttle -vvv --disable-ipv6 -r pi@rpi2.lan 0/0
Starting sshuttle proxy.
firewall manager: Starting firewall with Python version 3.5.0
firewall manager: ready method name pf.
IPv6 enabled: False
UDP enabled: False
DNS enabled: False
Binding redirector: 12300
TCP redirector listening on ('127.0.0.1', 12300).
TCP redirector listening with <socket.socket fd=6, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 12300)>.
Starting client with Python version 3.5.0
c : connecting to server...
c : executing: ['ssh', 'pi@rpi2.lan', '--', 'P=python3.5; $P -V 2>/dev/null || P=python; exec "$P" -c \'import sys; verbosity=3; stdin=getattr(sys.stdin,"buffer",sys.stdin); exec(compile(stdin.read(915), "assembler.py", "exec"))\'']
Enter passphrase for key '/Users/nkts/.ssh/id_rsa':
c :  > channel=0 cmd=PING len=7 (fullness=0)
server: assembling u'sshuttle' (8 bytes)
server: assembling u'sshuttle.cmdline_options' (27 bytes)
server: assembling u'sshuttle.helpers' (864 bytes)
server: assembling u'sshuttle.ssnet' (5500 bytes)
server: assembling u'sshuttle.hostwatch' (2334 bytes)
server: assembling u'sshuttle.server' (3089 bytes)
Starting server with Python version 2.7.9
 s: latency control setting = True
 s: available routes:
 s:   2/192.168.1.0/24
c : Connected.
<...>
```

**Comment by brianmay** [comment]
Just about to push a fix that I think should fix this. Just noticed that the error you got above will appear before the message I wanted is displayed. Oops.

**Comment by vieira** [comment]
Regarding ipv6 support with the method pf, I was trying to add support for it but hit a wall. There seems to be a bug in the kernel. When the packets are redirected the checksum is not calculated correctly and everything gets dropped by the kernel never reaching sshuttle.

[Bug 179392 - [pf] [ip6] Incorrect TCP checksums in rdr](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=179392#c16)

I did not test this in OSX yet, but as Apple forked the pf from FreeBSD it's likely that it has the same problem :disappointed:

**Comment by brianmay** [comment]
From the referenced report, it looks like this got fixed in FreeBSD version stable/10.

Just because Apple has forked pf from FreeBSD doesn't mean they haven't fixed the problem. I suspect ipv6 support in pf might be a low priority for them however.

**Comment by vieira** [comment]
Yeah, from the report it should be fixed on stable/10 but I am running 10.2 and the problem is still there as you can see from my last comment there. :disappointed: 

Regarding OSX, yes absolutely. They could have identified/fixed the problem on their own implementation. But I don't have access to any ipv6 network where I could connect a mac to test, right now.
